### DISCUSSION TITLE
useContractRead() inside a for loop: TypeError: Cannot read properties of undefined (reading 'current')

### DISCUSSION THREAD

**Discussion by davedavinci** [wevm/wagmi#3827] [Q&A]
I'm getting this error.
![image](https://github.com/wevm/wagmi/assets/163287064/8ba494c1-ecca-443b-94e6-2bfc3c9dd5df)
Code is like this:

```
  const {data: ecosystemTokens, isSuccess: succ} = useContractRead({
    ...raffleContract,
    functionName: 'getEcosystemTokens',
    watch:true,
  });
  if (succ) {
    for(let i=0; i < ecosystemTokens?.length; i++){
      const {data: tokenProp, isSuccess:gotData} = useContractRead({ ...raffleContract, functionName: 'getEcoTokenInfo', args:[ecosystemTokens[i]], watch:true})
      if(gotData){
        let token:EcoToken = {
          tokenAddress: tokenProp?.tokenAddress,
          tokenPair: tokenProp?.tokenPair,
          tokenPairType: tokenProp?.tokenPairType,
          tokenFeeTier: tokenProp?.tokenFeeTier,
          tokenName: tokenProp?.tokenName,
          tokenPoolType: tokenProp?.tokenPoolType,
          tokenWhitelisted: tokenProp?.tokenWhitelisted
        }
        if (token.tokenWhitelisted == true) spendable.push(token)
      }
    }
  }
```

The weird part is that it often doesn't occur.What can it be related to? Server side rendering?
Can someone tell if the above syntax is wrong or suggest where to investigate to solve this?
Thanks guys.

**Comment by hungdoansy** [comment]
Hooks shouldn't be used inside loop I think

**Reply by davedavinci** [reply]
Shall I drop WAGMI for this case then?
`ecosystemTokens` is an array of addresses. 
They are mapped in the contract to objects that have got all the properties I need:
![image](https://github.com/wevm/wagmi/assets/163287064/69d07291-5f51-4e00-8870-90d03c864c0f)
What would be the best practice in WAGMI to obtain that array of "whitelisted" objects?
Again, this is weird because it happens not every time. I've bee able to see this working, but there's somethingthat triggers this error more or less often and I can't understand what it is. Cheers! : )

**Comment by davedavinci** [comment]
I'm also trying to map every address to a hook:the error is gone, but this doesn't produce any result. Again, maybe I'm doing something wrong.

```
  if(succ) {
    ecosystemTokens?.map(address => {
      const {data: tokenProp, isSuccess:gotData} = useContractRead({ ...raffleContract, functionName: 'getEcoTokenInfo', args:[ecosystemTokens?.[address]], watch:true})
      if(gotData){
        let token:EcoToken = {
          tokenAddress: tokenProp?.tokenAddress,
          tokenPair: tokenProp?.tokenPair,
          tokenPairType: tokenProp?.tokenPairType,
          tokenFeeTier: tokenProp?.tokenFeeTier,
          tokenName: tokenProp?.tokenName,
          tokenPoolType: tokenProp?.tokenPoolType,
          tokenWhitelisted: tokenProp?.tokenWhitelisted
        }
        if (token.tokenWhitelisted == true) spendable.push(token)
      }
    })
  }
```

**Reply by hungdoansy** [reply]
Can you provide more codes?

**Reply by davedavinci** [reply]
I don't get it. What code do you need?
How would you interrogate contract in WAGMI if you have to do it with a parameterized query, depending on a loop?
